src/statmodels.jl

```diff
@@ -160,17 +160,50 @@ df_residual(obj::RegressionModel) = error("df_residual is not defined for $(type
 ## Nms are the coefficient names, corresponding to rows in the table
 type CoefTable
     mat::Matrix
+    cols::Tuple
     colnms::Vector
     rownms::Vector
-    pvalcol::Integer
-    function CoefTable(mat::Matrix,colnms::Vector,rownms::Vector,pvalcol::Int=0)
-        nr,nc = size(mat)
-        0 <= pvalcol <= nc || error("pvalcol = $pvalcol should be in 0,...,$nc]")
+    function CoefTable(cols::Tuple,colnms::Vector,rownms::Vector)
+        nc = length(cols)
+        nrs = map(length,cols)
+        nr = max(nrs...)
         length(colnms) in [0,nc] || error("colnms should have length 0 or $nc")
         length(rownms) in [0,nr] || error("rownms should have length 0 or $nr")
-        new(mat,colnms,rownms,pvalcol)
+        if all(unique(nrs) .== nr)
+            mat = hcat(cols...)
+        else
+            mat = Matrix(nr,nc) # A better value here?
+        end
+        new(mat,cols,colnms,rownms)
+    end
+end
+function CoefTable(mat::Matrix,colnms::Vector,rownms::Vector,pvalcol::Int=0)
+    nr,nc = size(mat)
+    cols = Any[mat[:, i] for i in 1:nc]
+    if pvalcol != 0                         # format the p-values column
+        cols[pvalcol] = PValue(cols[pvalcol])
+    end
+    cols = ntuple((i)->cols[i],nc)
+    CoefTable(cols,colnms,rownms)
+end
+
+type PValue
+    v::Number
+    function PValue(v::Number)
+        0. <= v <= 1. || error("p-values must be in [0.,1.]")
+        new(v)
+    end
+end
+function PValue(v::Array)
+    v2 = similar(v,PValue)
+    for i in eachindex(v)
+        v2[i] = PValue(v[i])
     end
+    v2
 end
+convert(::Type{PValue}, x::Number) = PValue(x)
+convert{T <: Number}(::Type{T}, x::PValue) = T(x.v)
+promote_rule{T<:Number}(::Type{T}, ::Type{PValue}) = T
```

Comment:
This can't be right. Maybe `promote_rule{S<:Number, T<:Number}(::Type{T}, ::Type{PValue{S}}) = PValue{promote_type(S, T)}`? But do we really need this? (Same for the second conversion above.)